### PR TITLE
Separate publish script into its own file, minor doc edits.

### PR DESCRIPTION
**Pull request by bdice**
## Description
Revises #7 and some assorted doc fixes found while reading the _rendered_ docs on ReadTheDocs (previously I only skimmed the source code).

## Checklist:
- [x] I agree with the terms of the [Contributor Agreement](https://github.com/glotzerlab/synced_collections/blob/main/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/synced_collections/blob/main/contributors.yaml).
- [ ] The changes introduced by this pull request are covered by existing or newly introduced tests.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/synced_collections/blob/main/changelog.txt) and added any related issue and pull request numbers for future reference.

### CODE REVIEW

File: synced_collections/data_types/synced_dict.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ class SyncedDict(SyncedCollection, MutableMapping):
     there are important distinctions to remember. In particular, because
     operations are reflected as changes to an underlying backend, copying (even
     deep copying) a :class:`SyncedDict` instance may exhibit unexpected
-    behavior. If a true copy is required, you should use the `_to_base()`
+    behavior. If a true copy is required, you should use the :meth:`~._to_base`
```
Comment (bdice) [inline synced_collections/data_types/synced_dict.py]: @vyasr Just verifying -- do we want to recommend `_to_base` or the call operator? The docs for the concrete classes recommend the call operator but `SyncedDict` and `SyncedList` recommend using `_to_base` directly. The difference is that the call operator does a `_load` first.

Comment (vyasr) [reply synced_collections/data_types/synced_dict.py]: Good catch. Definitely want to use the call operator. These docs have never been published, so this was probably missed when we originally reviewed and then we never updated it.

Comment (vyasr) [reply synced_collections/data_types/synced_dict.py]: Added docs of the call operator so that we can link to it.

File: changelog.txt [hunk 2]
```diff
@@ -6,7 +6,7 @@ The **synced_collections** package follows `semantic versioning <https://semver.
 
 
 [0.0.0] -- XXXX-XX-XX
----------------------
+=====================
```
Comment (vyasr) [inline changelog.txt:9]: Just FYI, to answer https://github.com/glotzerlab/synced_collections/pull/7#discussion_r1189909613, I removed the overall version headings to make the automated parsing of the changelog for the release easier. Don't think the headers add much anyway.

**Review by vyasr — APPROVED** [review]

**Pull request merged**